File: Lab_RGR/Lazy Initialization.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_RGR
{
    // Creational Pattern - Lazy Initialization
    public class PhotoManager
    {
        private List<string> _photoList;

        public List<string> PhotoList
        {
            get
            {
                if (_photoList == null)
                {
                    _photoList = new List<string>();
                }
                return _photoList;
            }
        }

        public void AddPhoto(string photo)
        {
            PhotoList.Add(photo);
            Console.WriteLine($"Photo added: {photo}");
        }

        public void DisplayPhotos()
        {
            Console.WriteLine("Photos:");
            foreach (var photo in PhotoList)
            {
                Console.WriteLine(photo);
            }
        }
    }
}

File: Python_Patterns/Lazy_Initialization.py
class PhotoManager:
    def __init__(self):
        self._photo_list = None

    @property
    def photo_list(self):
        if self._photo_list is None:
            self._photo_list = []
        return self._photo_list

    def add_photo(self, photo):
        self.photo_list.append(photo)
        print(f"Photo added: {photo}")

    def display_photos(self):
        print("Photos:")
        for photo in self.photo_list:
            print(photo)
